identity-wallet/src/state/search/reducers/add_recent_search.rs

```diff
@@ -0,0 +1,31 @@
+use crate::error::AppError;
+use crate::state::search::actions::add_recent_search::AddRecentSearch;
+use crate::state::{
+    actions::{listen, Action},
+    AppState,
+};
+
+const MAX_RECENT_SEARCHES: usize = 3;
+
+pub async fn add_recent_search(state: AppState, action: Action) -> Result<AppState, AppError> {
+    if let Some(recent_search) = listen::<AddRecentSearch>(action) {
+        let mut search_results = state.search_results.clone();
```

Comment:
No need to clone here :) `let mut search_results = state.search_results;` will simply take (mutable) ownership over `state.search_results`.